src/autocomplete.js

```diff
@@ -0,0 +1,193 @@
+  'october',
+  'november',
+  'december'
+]
+const month = months.join('|')
+
+const weekStart = `starting +the +week +on +${day}`
+const weekStarts = days.map(
+  dayValue => `starting the week on ${dayValue}`
+)
+
+const and = ',|and +'
+const ands = [', ', ' and ']
+
+const byWeekDay = `on +${day} +(${and} +${day})*`
+const byWeekDays = days.flatMap(
+  dayValue => [`on ${dayValue}`, ...ands.flatMap(
+    andValue => [`on ${dayValue}${andValue}`, ...days.map(
+      subDayValue => `on ${dayValue}${andValue} ${subDayValue}`
+    )]
+  )]
+)
+
+const byMonth = `in +${month} +(${and} +${month})*`
+const byMonths = months.flatMap(
+  monthValue => [`in ${monthValue}`, ...ands.flatMap(
+    andValue => [`in ${monthValue}${andValue}`, ...months.map(
+      subMonthValue => `in ${monthValue}${andValue} ${subMonthValue}`
+    )]
+  )]
+)
+
+const ext = 'st|nd|rd|th'
+const exts = ext.split('|')
+
+const monthDay = `\\d+ +${ext}`
+const monthDays = exts.map(
+  extValue => `\\d+ ${extValue}`
+)
+
+const byMonthDay = `the ${monthDay} (${and} +${monthDay})*`
+const byMonthDays = monthDays.flatMap(
+  monthDayValue => [`in ${monthDayValue}`, ...ands.flatMap(
+    andValue => [`in ${monthDayValue}${andValue}`, ...monthDays.map(
+      subMonthDayValue => `in ${monthDayValue}${andValue} ${subMonthDayValue}`
+    )]
+  )]
+)
+
+const yearDay = `\\d+ +${ext}`
+const yearDays = exts.map(
+  extValue => `\\d+ ${extValue}`
+)
+
+const byYearDay = `on +the +${yearDay} +${ext} +(${and} +${yearDay} +${ext})* +day +of +the +year`
+const byYearDays = yearDays.flatMap(
+  yearDayValue => [`on the ${yearDayValue}`, ...exts.flatMap(
+    extValue => [`on the ${yearDayValue} ${extValue}`, ...ands.flatMap(
+      andValue => [`on the ${yearDayValue} ${extValue}${andValue}`, ...yearDays.map(
+        subYearDayValue => `on the ${yearDayValue} ${extValue}${andValue} ${subYearDayValue}`
+      )]
+    )]
+  )]
+)
+
+const weekNo = '\\d+'
+
+const byWeekNo = `on +the +${weekNo} +${ext} +(${and} +${weekNo} +${ext})* +week +of +the +year`
+const byWeekNos = exts.flatMap(
+  extValue => [`on the ${weekNo} ${extValue} week of the year`, ...ands.flatMap(
+    andValue => [`on the ${weekNo} ${extValue}${andValue}`, ...yearDays.map(
+      yearDayValue => `on the ${weekNo} ${extValue}${andValue} ${yearDayValue} ${extValue} week of the year`
+    )]
+  )]
+)
+
+const hour = '[0-3]?\\dh'
+const minute = '[0-5]\\dm'
+const second = '[0-5]\\ds'
+const time = `(${hour})?(${minute})?(${second})?`
+const times = [
+  '\\d+h',
+  '\\d+h\\d+m',
+  '\\d+h\\d+m\\d+s',
+  '\\d+m',
+  '\\d+m\\d+s',
+  '\\d+s'
+]
+
+const byTime = `at +${time} +(${and} +${time})*`
+const byTimes = times.flatMap(
+  timeValue => [`at ${timeValue}`, ...ands.flatMap(
+    andValue => times.flatMap(
+      subTimeValue => [`at ${timeValue}${andValue} ${subTimeValue}`]
+    )
+  )]
+)
+
+const until = `until +month +\\d+ +, +\\d{4} +(at +${time})?`
+const untils = [
+  'until month \\d+, \\d+',
+  ...times.flatMap(
```

Comment:
You are using `flatMap` but you are not returning an array. Use `map` instead. Or `join` if that's what you intended.